Information_Gain_Arka.jl

```diff
@@ -0,0 +1,76 @@
+
+using Random
+using Statistics
+using StatsBase  # Importing StatsBase for the countmap function
+
+"""
+    entropy(y::Vector)
+
+Calculate the entropy of a vector of labels `y`.
+"""
+function entropy(y::Vector)
+    counts = countmap(y)
+    probs = values(counts) ./ length(y)
+    return -sum(p -> p > 0 ? p * log2(p) : 0, probs)
+end
+
+"""
+    information_gain(y::Vector, y_left::Vector, y_right::Vector)
+
+Calculate the Information Gain of a split. `y` is the original labels vector, `y_left` and `y_right` are the labels vectors
+for the left and right splits, respectively.
+"""
+function information_gain(y::Vector, y_left::Vector, y_right::Vector)
+    H = entropy(y)
+    H_left = entropy(y_left)
+    H_right = entropy(y_right)
+    p_left = length(y_left) / length(y)
+    p_right = length(y_right) / length(y)
+    return H - (p_left * H_left + p_right * H_right)
+end
+
+"""
+    split_dataset(X::Matrix, y::Vector, feature::Int, threshold::Real)
+
+Split the dataset `X` and labels `y` based on a `feature` and a `threshold`.
+Returns the left and right splits for both `X` and `y`.
+"""
+function split_dataset(X::Matrix, y::Vector, feature::Int, threshold::Real)
```

Comment:
does this handle non-numerical features?

Reply:
Yes. You can check the test file